tests/test_enhanced_sampling.py

```diff
@@ -120,6 +121,45 @@ def align_sample(x_gas, x_solvent):
     return xb_new
 
 
+from typing import List
+from md.states import CoordsVelBox
+
+
+class AlignedMover(MultipleTryMetropolis):
+    def __init__(self, K, vacuum_samples, vacuum_log_weights, log_prob_fn):
+
+        self.vacuum_samples = vacuum_samples
+        self.vacuum_log_weights = vacuum_log_weights
+        self.batch_log_prob_fn = jax.vmap(log_prob_fn)
+
+        super().__init__(K)
+
+    def batch_propose(self, x: CoordsVelBox) -> List[CoordsVelBox]:
+
+        ligand_samples = enhanced.sample_from_log_weights(self.vacuum_samples, self.vacuum_log_weights, self.K)
```

Comment:
Suggestion: Another way to split this up would be to have a `AlignedLigandResampler(samples, log_weights)` or similar that knows how to do `batch_propose`, and then pass the `resampler.batch_propose` function as an argument to the `MultipleTryMetropolis(batch_log_prob, batch_propose)` constructor.

Reply:
`AlignedLigandResampler` would also need to know the water coords to generate correct samples, and the `x_solvent` changes on each frame.

Reply:
Yes, it would have a `.batch_propose(x)` method, exactly as implemented here! Just it would be its own object that could "lend" out its `batch_propose` method to a `MultipleTryMetropolis` constructor, rather than being a definition that lives inside a `MultipleTryMetropolis` subclass.

Reply:
ah okay - I think I see what you're saying. I can probably make that work.

Reply:
(The same ligand-alignment functionality is used also in a postprocessing step you had implemented -- seems like the same functionality could be defined once and used in both places.)